src/hooks/useTxHistory.ts

```diff
@@ -11,16 +19,26 @@ const useTxHistory = (
   error?: string
   loading: boolean
 } => {
+  const router = useRouter()
+
   const { safe, safeAddress, safeLoaded } = useSafeInfo()
   const { chainId } = safe
 
   // If pageUrl is passed, load a new history page from the API
-  const [page, error, loading] = useAsync<TransactionListPage>(
+  const [page, error, loading] = useAsync<
+    | TransactionListPage
+    | SafeIncomingTransfersResponse
+    | SafeMultisigTransactionsResponse
+    | SafeModuleTransactionsResponse
```

Comment:
They are essentially the same list of tx items, aren't they? Can we create a wrapper around `getFilteredTxHistory` that would return a `TransactionListPage`?

Reply:
They are each a ['subset' of the `TransactionListPage` type](https://github.com/safe-global/safe-react-gateway-sdk/blob/main/src/types/transactions.ts#L235-L253).

Reply:
If they are a subset thereof, no need to create a union type here.